texastribune/salesforce-stripe
# Circle page

#### What's this PR do?
Creates a one-stop shop Circle page that contains both info about the program and the payment form.

#### Why are we doing this? How does it help us?
So we can kill the donations app! And to get the current Circle form off OG styles/JS. And because it just makes sense to have all Circle-related things living on one page.

#### How should this be manually tested?
+ Do a fresh `make interactive`
+ `yarn run dev`
+ Go to `/circleform`. Confirm that Editor's Circle, monthly, is selected.
+ Ensure form validation is working as expected. You could do any of the following: not fill out first or last name; put in an invalid email address; put in a zip code that has letters and/or is not 5 characters long; put in real card information. You should not be able to submit and should get meaningful error messages.
+ Put in a [test card number](https://stripe.com/docs/testing#cards) and confirm you get the thank-you page.
+ Visit `/circleform?level=<level>` with the following values. Ensure the proper radio button is selected by default on load. `editorMonthly`, `editorYearly`, `leadershipMonthly`, `leadershipYearly`, `chairmanMonthly`, `chairmanYearly`.

#### Have automated tests been added?
No.

#### Has this been tested on mobile?
Yes.

#### Are there performance implications?
No.

#### What are the relevant tickets?
https://3.basecamp.com/3098728/buckets/7046813/todos/1010306801

#### How should this change be communicated to end users?
NA.

#### Next steps?
Blast form.

#### Smells?
NA.

#### TODOs:
* [x] Get darkened background image
* [x] Art sign-off
* [x] Evan sign-off
* [x] Emily sign-off
* [x] Terry sign-off

#### Has the relevant documentation/wiki been updated?
I'll be doing comprehensive docs for this whole rewrite when finished.

#### Technical debt note
Less.
